src/octree/node.rs

```diff
@@ -469,8 +469,12 @@ impl Drop for NodeWriter {
 }
 
 impl NodeWriter {
-    pub fn new(octree_meta: &OctreeMeta, node_id: &NodeId) -> Self {
-        let stem = node_id.get_stem(&octree_meta.directory);
+    pub fn new(
+        octree_data_provider: &OnDiskOctreeDataProvider,
```

Comment:
i'd add a comment about NodeWriter being implemented only for OnDiskDataProvider